concent_api/core/payments/sci_backend.py

```diff
@@ -0,0 +1,95 @@
+import math
+from decimal        import Decimal
+from django.conf    import settings
+from golem_messages import message
+from golem_sci      import implementation
+
+
+def get_number_of_eth_block(concent_rpc: implementation.SCIImplementation):
+    assert isinstance(concent_rpc, implementation.SCIImplementation)
+    return concent_rpc.get_block_number()
+
+
+def get_list_of_payments(
+    concent_rpc:            implementation.SCIImplementation    = None,
+    requestor_eth_address:  str                                 = None,
+    provider_eth_address:   str                                 = None,
+    payment_ts:             int                                 = None,
+    current_time:           int                                 = None,
+    to_block:               int                                 = None,
+    transaction_type:       str                                 = None,
+):
+    """
+    Function which return list of transactions from payment API
+    where timestamp >= T0
+    """
+    assert isinstance(concent_rpc,              implementation.SCIImplementation)
+    assert isinstance(requestor_eth_address,    str) and len(requestor_eth_address) == 42
+    assert isinstance(provider_eth_address,     str) and len(provider_eth_address)  == 42
+    assert isinstance(payment_ts,               int) and payment_ts     > 0
+    assert isinstance(current_time,             int) and current_time   > 0
+    assert isinstance(to_block,                 int) and to_block       > 0
+    assert isinstance(transaction_type,         str) and transaction_type in ['Force', 'Batch']
+
+    chain_segment_amount    = math.ceil((current_time - payment_ts) / settings.CREATION_NEW_CHAIN_SEGMENT_TIME)
+    from_block              = to_block - chain_segment_amount
+
+    if transaction_type == 'Force':
```

Comment:
please extract thi `if ... else` to a separate function

Reply:
I was wondering if it is necessary? This `get_list_of_payments` is a function for getting lists of all transactions from geth. If I extract `if ... elif` to separate function then in this block of code will be only asserts.